packages/ui/src/components/Datepicker/Datepicker.tsx

```diff
@@ -82,21 +82,23 @@ export interface DatepickerRef {
   clear: () => void;
 }
 
-export interface DatepickerProps extends Omit<TextInputProps, "theme"> {
+export interface DatepickerProps extends Omit<TextInputProps, "theme" | "onChange"> {
+  defaultDate?: Date;
   open?: boolean;
   inline?: boolean;
   autoHide?: boolean;
   showClearButton?: boolean;
   labelClearButton?: string;
   showTodayButton?: boolean;
   labelTodayButton?: string;
-  defaultDate?: Date;
   minDate?: Date;
   maxDate?: Date;
   language?: string;
   weekStart?: WeekStart;
   theme?: DeepPartial<FlowbiteDatepickerTheme>;
-  onSelectedDateChanged?: (date: Date) => void;
+  onChange?: (date: Date) => void;
+  customValue?: Date | null;
```

Comment:
Is `customValue` a good name? (non-blocking)

Reply:
suggestions ?

Reply:
@SutuSebastian @tulup-conner

Reply:
I'd keep only:

```ts
value?: Date;
defaultValue?: Date;
format?: string | (value: Date) => string; // date format | custom final value
```

Reply:
Yeah, I think that I agree, with `value` and `defaultValue`.

Reply:
@SutuSebastian Sorry, I guess I didn't understand very well.

So 
customValue = value 
defaultdDate = defaultValue 
format = ?????

Reply:
forget about `customValue`, since thats what `format` is for, it acts like a final resolver before rendering the value inside the input to be displayed

Reply:
these follow the vanilla pattern of any `<input />` component

```ts
value?: Date;
defaultValue?: Date;
```

Reply:
>If so, can it be done in another PR please?

and yes, it can be done in another PR

Reply:
So, would renaming 

customValue to value
defaultDate to defaultValue

work in this case?

@SutuSebastian @rluders

Reply:
Yep
Keep as much default behaviour as possible

Reply:
Okay, now I recall why I've used `customValue`, here is why:

If you use value, React re-rendering mechanism is more consistent by using the standard input value, triggering the lifecycle even if the value isn't passed at all.

That means negative use cases as: DefaultEmpty / NullDateValue / EmptyDates will not be achievable with that specific logic, considering the previous component logic.

That means:

1 - You still can use `value` as much as you want
2 - `customValue` can overwrite value just in case you need to change Manually ( and that's what the PR stands for )

Reply:
Can you provide an example please ?

Reply:
```ts
// with parsed value derived from `value` state
function DemoDerivedFromState() {
  const [value, setValue] = useState<Date | null>(new Date())

  return <DatePicker value={value} format={value?.toISOString()} onChange={setValue} />
}

// with parsed value derived `format` prop internal `value` argument
function DemoDerivedFromProp() {
  const [value, setValue] = useState<Date | null>(new Date())

  return <DatePicker value={value} format={(value) => value?.toISOString()} onChange={setValue} />
}

```

Reply:
@SutuSebastian  added on storybook a controlled version of the component to the templates so we can have the options controlled/uncontrolled after  renaming/removing customValue / defaultDate.

Thanks.